File: frontend/types/schedule.ts
export type ScheduleEvent = {
  id?: string;
  title: string;
  start_time: string;
  end_time: string;
  location?: string;
  notes?: string;
  notified?: boolean;
<<<<<<< HEAD
=======
  completed?: boolean;
>>>>>>> a78aa0fd5a16184ec5ef421650b3c03395164c66
};

export type Schedule = {
  id: string;
  user_id: string;
  events: ScheduleEvent[];
  created_at: string;
  updated_at: string;
};

export type ScheduleUploadResponse = {
  extracted_text: string;
  events: ScheduleEvent[];
  is_valid_schedule: boolean;
  message?: string;
};

File: backend/app/schemas/schedule.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict

class ScheduleEventBase(BaseModel):
    title: str
    start_time: str
    end_time: str
    location: Optional[str] = None
    notes: Optional[str] = None
    notified: bool = False
    completed: bool = False

class ScheduleEventCreate(ScheduleEventBase):
    pass

class ScheduleEventUpdate(ScheduleEventBase):
    title: Optional[str] = None
    start_time: Optional[str] = None
    end_time: Optional[str] = None
    notified: Optional[bool] = None

class ScheduleEventResponse(ScheduleEventBase):
    id: Optional[str] = None

class ScheduleResponse(BaseModel):
    id: str
    user_id: str
    events: List[ScheduleEventResponse]
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

class ScheduleUploadResponse(BaseModel):
    extracted_text: str
    events: List[ScheduleEventCreate]
    is_valid_schedule: bool = True
    message: Optional[str] = None
